### PR TITLE
Fix google tag manager

### PR DESCRIPTION
**Pull request by bumi**
thanks to @stephanpavlovic

### CODE REVIEW

**Comment by bumi** [comment]
@stephanpavlovic what was the final result of your test? did it work for you?

File: lib/rack/tracker/google_tag_manager/google_tag_manager.rb [hunk 1]
```diff
@@ -10,10 +10,10 @@ def inject(response)
     # Sub! is enough, in well formed html there's only one head or body tag.
     # Block syntax need to be used, otherwise backslashes in input will mess the output.
     # @see http://stackoverflow.com/a/4149087/518204 and https://github.com/railslove/rack-tracker/issues/50
-    response.sub! %r{<head>} do |m|
+    response.sub! %r{<head.*>} do |m|
       m.to_s << self.render_head
     end
-    response.sub! %r{<body>} do |m|
+    response.sub! %r{<body.*>} do |m|
```
Comment (DonSchado) [inline lib/rack/tracker/google_tag_manager/google_tag_manager.rb]: 👍

File: spec/integration/google_tag_manager_integration_spec.rb [hunk 2]
```diff
@@ -5,15 +5,21 @@
...
 
   it "embeds the script tag with tracking event from the controller action" do
+    visit '/'
     expect(page.find("head")).to have_content 'GTM-ABCDEF'
-    expect(page.find("head")).to have_content "dataLayer.push( {\"click\":\"X\",\"price\":10}, {\"transactionProducts\":[{\"sku\":\"DD44\",\"name\":\"T-shirt\"},{\"sku\":\"DD66\",\"name\":\"Jeans\"}]} );"
+    expect(page.find("head")).to have_content "dataLayer.push({\"click\":\"X\",\"price\":10}, {\"transactionProducts\":[{\"sku\":\"DD44\",\"name\":\"T-shirt\"},{\"sku\":\"DD66\",\"name\":\"Jeans\"}]});"
     expect(page.find("body")).to have_xpath '//body/noscript/iframe[@src="https://www.googletagmanager.com/ns.html?id=GTM-ABCDEF"]'
   end
 
+  it "does not inject a dataLaye if no events are set " do
```
Comment (DonSchado) [inline spec/integration/google_tag_manager_integration_spec.rb]: dataLaye_r_

File: spec/support/metal_controller.rb [hunk 3]
```diff
@@ -48,9 +48,11 @@ def google_analytics
   end
 
   def google_tag_manager
-    tracker do |t|
-      t.google_tag_manager :push, { click: 'X', price: 10 }
-      t.google_tag_manager :push, transactionProducts: [{ sku: 'DD44', name: 'T-shirt' }, { sku: 'DD66', name: 'Jeans' }]
+    unless params[:no_events]
```
Comment (DonSchado) [inline spec/support/metal_controller.rb]: hm.. I would like to avoid logic here

Comment (bumi) [inline spec/support/metal_controller.rb]: hmmm. it is just the spec. and I thought better than duplicating the action. hmmm...

**Pull request merged**